src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -427,7 +427,10 @@ class ActivityFeed extends React.Component<Props, State> {
         const { file } = this.props;
         const { restored_from, modified_at, file_version } = file;
 
-        if (restored_from) {
+        // Ensures restored version is only added on first feed load
+        const hasRestoredVersion = versions.entries.find((version) => version.action === VERSION_RESTORE_ACTION);
```

Comment:
Is there a better way you can detect this rather than looping through all the versions?

Reply:
^ looking at the last element in the versions array instead now :)